flax/linen/attention.py

```diff
@@ -338,13 +313,46 @@ def __call__(self,
     return out
 
 
-class SelfAttention(MultiHeadDotProductAttention):
-  """Self-attention special case of multi-head dot-product attention."""
+class MultiHeadDotProductAttention(_BaseMultiHeadDotProductAttention):
+  """Multi-head dot-product attention.
 
+    Attributes:
+      num_heads: number of attention heads. Features (i.e. inputs_q.shape[-1])
```

Comment:
Why don't we put the attribute descriptions on the base class? I think that these should appear correctly then if you are using an auto-suggestion tool. Seems to be a bit odd to put them here, or did you have a reason for that?

Looking at RTD for `SelfAttention` now doesn't have the descriptions of the attributes: https://flax--1962.org.readthedocs.build/en/1962/_autosummary/flax.linen.SelfAttention.html#flax.linen.SelfAttention

Reply:
Done as requested.  Just wondering though: will these show up in the documentation?  If not, we could make two copies of the arguments.

Reply:
Did you know that you can view the documentation in your change by going to the bottom of the conversations tab and clicking on more details in the  "docs/readthedocs.org:flax" check?

This is `SelfAttention` in your branch: https://flax--1962.org.readthedocs.build/en/1962/_autosummary/flax.linen.SelfAttention.html#flax.linen.SelfAttention